src/game/inject.c

```diff
@@ -487,24 +491,24 @@ static void Inject_TextureData(
 
     for (int i = 0; i < inj_info->sprite_count; i++) {
         GAME_OBJECT_ID object_num;
+        int16_t num_meshes;
+        int16_t mesh_index;
         File_Read(&object_num, sizeof(int32_t), 1, fp);
+        File_Read(&num_meshes, sizeof(int16_t), 1, fp);
+        File_Read(&mesh_index, sizeof(int16_t), 1, fp);
+
         if (object_num < O_NUMBER_OF) {
-            File_Read(&g_Objects[object_num], sizeof(int16_t), 1, fp);
-            File_Read(
-                &g_Objects[object_num].mesh_index, sizeof(int16_t), 1, fp);
-            g_Objects[object_num].mesh_index += level_info->sprite_info_count;
-            level_info->sprite_info_count -= g_Objects[object_num].nmeshes;
-            g_Objects[object_num].loaded = 1;
-        } else {
-            int32_t static_num = object_num - O_NUMBER_OF;
-            File_Skip(fp, 2);
-            File_Read(
-                &g_StaticObjects[static_num].mesh_number, sizeof(int16_t), 1,
-                fp);
-            g_StaticObjects[static_num].mesh_number +=
-                level_info->sprite_info_count;
-            level_info->sprite_info_count++;
+            OBJECT_INFO *object = &g_Objects[object_num];
+            object->nmeshes = num_meshes;
+            object->mesh_index = mesh_index + level_info->sprite_info_count;
+            object->loaded = 1;
+        } else if (object_num - O_NUMBER_OF < STATIC_NUMBER_OF) {
+            STATIC_INFO *object = &g_StaticObjects[object_num - O_NUMBER_OF];
+            object->nmeshes = num_meshes;
+            object->mesh_number = mesh_index + level_info->sprite_info_count;
+            object->loaded = true;
```

Comment:
This field is part of a bit field, so it still has to be `1` I think. Not sure how C will handle `true` in this case.

Reply:
This is the new flag I added to `STATIC_INFO` as a bool. I wanted to change `OBJECT_INFO` to match then realised it was a bit field there.

Reply:
I think we should remove bit fields altogether, as they're about optimizing for memory consumption and IMO we need code readability first these days.

Reply:
Yes, agreed, it would be much more readable. I can follow that up separately throughout.

Reply:
Yes – definitely a separate PR, not in scope of this.